tests: drivers: disk: Replace the MAX function with if and else

hunterhu3000 opened the pull request:
As the parameter of "MAX" is uint32, the nagetive number will be bigger than 0. The MAX function does not work as expected. So change it to if/else form.

Fixes #51146

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

gmarull commented on tests/drivers/disk/disk_access/src/main.c:
```diff
@@ -147,7 +150,11 @@ static void test_sector_write(uint8_t *wbuf, uint8_t *rbuf, uint32_t num_sectors
+	if ( disk_sector_count / 2 > num_sectors )
+                sector = disk_sector_count / 2 - num_sectors;
+        else
+                sector = 0;
```
all if/elses need `{}`, also please don't mix tabs and spaces...

hunterhu3000 replied:
Yes, CI found that problem.
I have updated the format. 
Thanks

hunterhu3000 committed:
tests: drivers: disk: Replace the MAX function with if and else

As the parameter of "MAX" is uint32, the nagetive number will be
bigger than 0. The MAX function does not work as expected. So
change it to if/else form.

Fixes #51146

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

hunterhu3000 commented:
All CI checks are passed, thanks to review.
